### ISSUE TITLE
[Question] How to customize loss?

### ISSUE THREAD

**Issue by MetallicaSPA** [DLR-RM/stable-baselines3#1537]
### ❓ Question

Hello, I'm using stable baselines 3 for a neurosymbolic reinforcement learning thesis. I want to compare the neurosymbolic reinforcement learning system with a RL system. I feel stable baselines 3 it's pretty straightforward to use so I made some experiments with it and gymnasium and so far so good.
My problem is that I want to calculate an expected reward from the symbolic system and use it as a ground truth to compare with the obtained reward. So I need to use my expected reward, not the one that the algorithms provides me.
Is there any way to do it with stable baselines? Thank you in advance.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
>  So I need to use my expected reward, not the one that the algorithms provides me

The algorithm doesn't provide any reward, the env does. Or you meant value function estimation?

>  I want to compare the neurosymbolic reinforcement learning system with a RL system.

For anything really custom, best is to fork SB3 and take a look at the code (we have a developer guide that will tell you where to start).

**Comment by MetallicaSPA** [comment]
> The algorithm doesn't provide any reward, the env does. Or you meant value function estimation?

 Yes, I mean that. I calculated the reward (an estimated reward based on the rules I defined in a symbolic way) so I want to compare that estimation with the one obtained.

I have checked the a2c.py file and under the train function I found this:
`
# Policy gradient loss
            policy_loss = -(advantages * log_prob).mean()

            # Value loss using the TD(gae_lambda) target
            value_loss = F.mse_loss(rollout_data.returns, values)

            # Entropy loss favor exploration
            if entropy is None:
                # Approximate entropy when no analytical form
                entropy_loss = -th.mean(-log_prob)
            else:
                entropy_loss = -th.mean(entropy)

            loss = policy_loss + self.ent_coef * entropy_loss + self.vf_coef * value_loss
`

Let's say I want to introduce a new variable to the loss calculation, an external one, that I have calculated in a neurosymbolical way, which will be a number. How can I introduce it so, for example, i can get this:

`loss = policy_loss + self.ent_coef * entropy_loss + self.vf_coef * value_loss + neuro_reward`

I'm wondering if when I do the call to the enviroment i have to add also the reward too; so instead of calling only the enviroment I'll have to add a call to that reward.

Thanks in advance.

**Comment by araffin** [comment]
>  How can I introduce it so, for example, i can get this:

as I wrote, for that you need a fork of SB3.

**Comment by MetallicaSPA** [comment]
Okay so I have found a some kind of solution. I'll explain it here just in case someone will have the same problem.
